src/cpp/session/modules/SessionPackages.R

```diff
@@ -23,531 +23,176 @@
 # map the repository string to the directory containing produced output
 .rs.setVar("availablePackagesPendingEnv", new.env(parent = emptyenv()))
 
-# a vectorized function that takes any number of paths and aliases the home
```

Comment:
Sorry, most of the changes within are refactoring -- I'll point out the actual code changes via comments below.